### DISCUSSION TITLE
How to show new project wizard by default

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/eclipse.platform.ui#500] [Q&A]
There is the following wiki page:

https://wiki.eclipse.org/FAQ_How_do_I_add_my_wizard_to_the_New,_Import,_or_Export_menu_categories%3F

but it is missing an the information how to add it directly to the menu (see the following screenshot):

![grafik](https://user-images.githubusercontent.com/1331477/210139071-6db51a03-83cd-4dbc-9885-1957d6c9433c.png)

**Comment by mickaelistria** [comment]
This is a perspective-specific extension: https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fextension-points%2Forg_eclipse_ui_perspectiveExtensions.html&cp%3D2_1_1_196&anchor=e.newWizardShortcut . I don't think the new menu allows typical extensions with org.eclipse.ui.menus extension point, but that's worth double-checking. Hopefully I'm wrong and you'd just need to find the right locationUri to use.

**Reply by laeubi** [reply]
> This is a perspective-specific extension

Great will try that, can you probably add that information to the wikipage? I already though it must be something like this but can't seem to find the right "connection" ...

> I don't think the new menu allows typical extensions with org.eclipse.ui.menus extension point, but that's worth double-checking. Hopefully I'm wrong and you'd just need to find the right locationUri to use.

I'm already able to put some content there (I created the screenshot that way) but it is:

1. a bit cumbersome to find the "right" URI/position it appropriately
2. then I need to handle all the wizard logic (fetching, creating dialog, perspective switches, selections, ...) manually, that don't feel right

**Reply by laeubi** [reply]
I now tried this out but getting hit by:
- https://github.com/eclipse-platform/eclipse.platform/issues/306

the items are only shown if I select `File` > `New` but not when I right click the project ...

**Reply by kjsmita6** [reply]
The menu URI (project explorer) we have is `popup:common.new.menu?before=new`, although I'm not sure about the `before=new` part. But the menu ID is definitely `common.new.menu`.

**Comment by laeubi** [comment]
I think that probably https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.isv/reference/extension-points/org_eclipse_ui_navigator_navigatorContent.html?cp=2_1_1_193#e.commonWizard what I'm looking for... I must confess that it is really confusing with all those different extension points and also for the user that items that seem similar appear at different locations.

**Reply by laeubi** [reply]
This already shows the wizard in the right menu, but the ordering is a bit odd I have opened:
- https://github.com/eclipse-platform/eclipse.platform.ui/issues/502

**Reply by akurtakov** [reply]
I hope to make things a bit less confusing for June release by finishing https://github.com/eclipse-platform/eclipse.platform.ui/commit/3ace015b981ec72599445d7fd2ee6afdaf244b63 . I agree that a lot of pruning is needed to make things less cofusing.
